src/main/java/carpet/mixins/TradeOutputSlot_scarpetEventMixin.java

```diff
@@ -0,0 +1,33 @@
+package carpet.mixins;
+
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.screen.slot.TradeOutputSlot;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.village.Merchant;
+import net.minecraft.village.TradeOffer;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+import static carpet.script.CarpetEventServer.Event.PLAYER_TRADES;
+
+@Mixin(value = TradeOutputSlot.class)
+public abstract class TradeOutputSlot_scarpetEventMixin {
+    @Shadow @Final private Merchant merchant;
+
+    @Inject(method = "onTakeItem", require = 0, at = @At(value = "INVOKE",
```

Comment:
why did you make is require=0? I would skip that and assume it successful. This way if mojang changes something we will immediately know, or have a chance to know.